### ISSUE TITLE
An error occurred when installing

### ISSUE THREAD

**Issue by simon-ritchie** [krassowski/jupyterlab-lsp#7]
A UnicodeDecodeError occurred when installing with the following command:

```
$ jupyter labextension install "@krassowski/jupyterlab-lsp"
```

```
Building jupyterlab assets
/Exception in thread Thread-2:

Traceback (most recent call last):
  File "c:\users\****\anaconda3\lib\threading.py", line 917, in _bootstrap_inner
    self.run()
  File "c:\users\****\anaconda3\lib\threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "c:\users\****\anaconda3\lib\subprocess.py", line 1238, in _readerthread
    buffer.append(fh.read())
UnicodeDecodeError: 'cp932' codec can't decode byte 0xef in position 7095: illegal multibyte sequence
```

# Environment

- OS: Windows 7（Japanese）
- jupyterlab==1.0.2
- conda-forge/win-64::conda-package-handling-1.4.1-py37_0
- conda-forge/win-64::nodejs-12.8.1-0
- conda-4.7.11-py37_0
- Python 3.7.3

# Notes

- `cp932` is used in the standard output of Japanese Windows computers (mainly command prompt, like anaconda prompt). So probably this error will not be occurred English environment or other OS, like Ubuntu.
- This error happen both on the command line and JupyterLab's Extension Manager.

# Temporary solution

When I use `--debug` argument to inspect, then build completed successfully (probably the change of the standard output contents during the build seems to solve this problem).

```
$ jupyter labextension install "@krassowski/jupyterlab-lsp" --debug
```

**Comment by krassowski** [comment]
Thanks for the detailed report and a workaround!

Unfortunately, the traceback is not very informative (everything seems to happen in the Python standard library). Also, this extension has no Python code, so it is not a simple "wrong encoding of a .py file" issue :)

**Comment by simon-ritchie** [comment]
Hi, @krassowski 
Thx for reply!

> Also, this extension has no Python code, so it is not a simple "wrong encoding of a .py file" issue :)

Ah...that's right.  
Maybe it needs to fix code related to build on the JupyterLab side.  
Since it is not an issue of this extension, I will close the issue.

Thx.

**Comment by krassowski** [comment]
Other than the installation issue, does the extension work well for you?

**Comment by simon-ritchie** [comment]
@krassowski 

Yes, extension worked well (amazing works!!).  

> automatic signature suggestions
> ...
> auto invocation of completer on certain characters, for example '.' (dot) in Python

Wow... although they were not implemented yesterday, but  I was surprised that they were already implemented  today!

Because I had been looking alternative of [Hinterland](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/hinterland) for a long time, I'm very happy to finally find an extension that could alternate it.

# Additional information (maybe trivial)

## About node.js installation instructions

The following error occurred during the initial installation: 

```
Please install nodejs >=6.11.5 before continuing. nodejs may be installed using conda or directly from the nodejs website.
```

Therefore, I downloaded node.js from the official website and installed it.

- Downloaded from: https://nodejs.org/en/
- Downloaded version: 10.16.3 LTS for Windows(x64)

However, as a result, the installation failed with the following error:

```
"@krassowski/jupyterlab-lsp" is not a valid npm package
```

So, I installed node.js via conda command, as followed, and installation of extension was succeeded:

```
$ conda install -c conda-forge nodejs
```

ref: [ValueError: "@jupyter-widgets/jupyterlab-manager" is not a valid npm package](https://github.com/jupyter-widgets/ipywidgets/issues/1982)

It maybe helpful if node.js and conda commands etc. are explained in `Prerequisites` of README.

## Pyflakes seems to be an error even in the command execution cell

 (very trivial proposal)

![20190822_1](https://user-images.githubusercontent.com/25178045/63513428-23662480-c521-11e9-8801-b4e520e34e6d.png)

It seems that code cells using the `!` symbol are also checked.  
If possible, it may be preferable to ignore like magic command cells (e.g., `%matplotlib inline`).

---

My impression is that it's a great extension! :D  
Perhaps I'll introduce this extension at tech blog.

Thx.

**Comment by krassowski** [comment]
Thanks for the feedback! The lines starting with `!` symbol are now excluded from linting in Python notebooks (totally forgot about it thanks for letting me know) - in just released 0.4.0 version.

Re developement speed: I need all this features too! ;)

**Comment by simon-ritchie** [comment]
Amazing!
Thank you for the very fast update! :D
